src/Symfony/Component/FeatureFlag/FeatureChecker.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    private array $cache = [];
+
+    public function __construct(
+        private readonly FeatureRegistry $featureRegistry,
+        private readonly mixed $default,
```

Comment:
Sure! It will cover most use cases.

Reply:
What is the use-case for a different default value?

Reply:
We could use "true" to allow access when a feature does not exist. This could be useful when opening a feature to everyone simply by removing the feature itself, not the checks.

It will also allow people to use Enum as return value (FeatureValue::Grant vs. FeatureValue::Deny or FeatureVersion::v1 vs. FeatureVersion::v2, etc.)

Reply:
> We could use "true" to allow access when a feature does not exist. This could be useful when opening a feature to everyone simply by removing the feature itself, not the checks.

This looks like a custom use case that should be covered for the final app by decorating the feature checkers chain. I would keep it simple by letting the exception run, so we cover other use cases.

Reply:
> We could use "true" to allow access when a feature does not exist. This could be useful when opening a feature to everyone simply by removing the feature itself, not the checks.

This default value is shared for all the features covered by this FeatureChecker. I don't see how changing the default can be used to set a static value for a single feature.

Support for ["static" feature](https://github.com/symfony/symfony/pull/53213/files#r1439891526) would solve this use-case.

Reply:
> This looks like a custom use case that should be covered for the final app by decorating the feature checkers chain. I would keep it simple by letting the exception run, so we cover other use cases.

I agree to throw an exception.

> This default value is shared for all the features covered by this FeatureChecker. I don't see how changing the default can be used to set a static value for a single feature.
>
> Support for ["static" feature](https://github.com/symfony/symfony/pull/53213/files#r1439891526) would solve this use-case.

I was thinking of a global solution : delete the feature to open it to everyone.


You are both right. We could throw an exception but, to avoid 500 errors due to a typo, we can make it configurable. 
Something like `$onInvalid` to throw an exception or just return `false` (and maybe log it or dispatch an event) ?

Reply:
> We could throw an exception but, to avoid 500 errors due to a typo, we can make it configurable.

Generally, we don't want to hide such error.